src/librustdoc/clean/utils.rs

```diff
@@ -421,29 +421,26 @@ crate fn resolve_type(cx: &DocContext<'_>, path: Path, id: hir::HirId) -> Type {
         Res::SelfTy(..) | Res::Def(DefKind::TyParam | DefKind::AssocTy, _) => true,
         _ => false,
     };
-    let did = register_res(&*cx, path.res);
+    let did = register_res(cx, path.res);
     ResolvedPath { path, param_names: None, did, is_generic }
 }
 
 crate fn get_auto_trait_and_blanket_impls(
-    cx: &DocContext<'tcx>,
+    cx: &mut DocContext<'tcx>,
     ty: Ty<'tcx>,
     param_env_def_id: DefId,
 ) -> impl Iterator<Item = Item> {
-    let auto_impls = cx
-        .sess()
-        .prof
+    let prof = &cx.tcx.sess.prof;
```

Comment:
What is the difference between these? Is it so the borrow-checker can see we're only using `.tcx.sess.prof` rather than all of `cx`?

Reply:
Yes, exactly. `cx.sess()` takes a mutable borrow because the `&self` in the function doesn't cause the reference to decay to a &T soon enough I think (??) it gave me errors as if it thought all of `&mut self` was borrowed and not just the immutable reference. Let me find the original errors, one sec.

Reply:
Ok my explanation above is wrong - the issue is `AutoTraitFInder::new` takes `&mut DocContext` but using `.sess()` ties the lifetime of the profiler to `DocContext` even though it shouldn't. Using `cx.tcx.sess` copies `tcx` instead and gives the session a lifetime of `'tcx`.

Reply:
A bit over my head, but seems harmless :)